storage/src/storage/sqlite/sync.rs

```diff
@@ -0,0 +1,923 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::convert::TryInto;
+
+use rusqlite::{params, OptionalExtension, Row, ToSql};
+use snarkvm_dpc::{AleoAmount, MerkleRootHash, Network, PedersenMerkleRootHash, ProofOfSuccinctWork};
+use tracing::*;
+
+#[cfg(feature = "test")]
+use crate::key_value::KeyValueColumn;
+use crate::{
+    BlockFilter,
+    BlockOrder,
+    CanonData,
+    FixMode,
+    SerialRecord,
+    SerialTransaction,
+    SyncStorage,
+    ValidatorError,
+};
+
+use super::*;
+
+fn read_static_blob<const S: usize>(row: &Row, index: usize) -> rusqlite::Result<[u8; S]> {
+    (&row.get::<_, Vec<u8>>(index)?[..])
+        .try_into()
+        .map_err(|_| rusqlite::Error::InvalidQuery)
+}
+
+impl SqliteStorage {
+    const SCHEMA_INDEX: u32 = 2;
+
+    fn migrate(&self, from: u32) -> Result<()> {
+        if from == 0 {
```

Comment:
what is the significance of these indices? they could use some comments

Reply:
the names should be self-describing as to their purpose (i.e. lookup transaction id). any particular confusion?

Reply:
I meant `from`, `SCHEMA_INDEX` and how they relate to the migration